gradle-diff-task-dependencies/plugin/src/main/groovy/com/novoda/gradle/diffdependencytask/ConditionalDependency.groovy

```diff
@@ -0,0 +1,33 @@
+package com.novoda.gradle.diffdependencytask
+
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+import java.util.regex.Pattern
+
+public class ConditionalDependency {
+
+    static {
+        Project.metaClass.registerConditionalDependency << { ConditionalDependency conditionalDependency ->
```

Comment:
I'm confused. If you want to add a method why not using the extension properties directly (ie: `project.ext`)? `Project` and `Task` are subtypes of [`ExtensionAware`](https://docs.gradle.org/current/dsl/org.gradle.api.plugins.ExtensionAware.html)